pkgs/development/python-modules/chex/default.nix

```diff
@@ -0,0 +1,54 @@
+{ absl-py
+, buildPythonPackage
+, dm-tree
+, fetchFromGitHub
+, jax
+, jaxlib
+, lib
+, numpy
+, toolz
+}:
+
+buildPythonPackage rec {
+  pname = "chex";
+  # As of 2021-12-29, the latest official version has broken tests with jax 0.2.26.
+  version = "unstable-2021-12-16";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "deepmind";
+    repo = pname;
+    rev = "5adc10e0b4218f8ec775567fca38b68bbad42a3a";
+    sha256 = "00xib6zv9pwid2q7wcr109qj3fa3g3b852skz8444kw7r0qxy7z3";
+  };
+
+  propagatedBuildInputs = [
+    absl-py
+    dm-tree
+    jax
+    numpy
+    toolz
+  ];
+
+  pythonImportsCheck = [
+    "chex"
+  ];
+
+  checkInputs = [
+    jaxlib
+  ];
+
+  # Flaky tests with ~25% flakiness caused by `set_n_cpu_devices` usage and error messages
+  # `RuntimeError: Attempted to set 1 devices, but 6 CPUs already available: ensure that `set_n_cpu_devices` is executed before any JAX operation.`
```

Comment:
Does this flakiness affect the test suite running outside of the nix environment as well? Is upstream aware of this issue?

Reply:
I couldn't reproduce it outside of nix env. Also when running via `pytest` (= after adding `pytestCheckHook`) I cannot reproduce it inside nix env either (ran multiple tens of builds without an issue). Given upstream uses `pytest` themselves when running the tests - it's unlikely they'll accept a bug report about `setup.py test` execution. Added `pytestCheckHook` into the derivation and removed tests exclusions.

Reply:
Sounds good, IIUC `pytestCheckHook` is the recommended way of running tests in nixpkgs now as well.